viewer/src/repository/cad/Repository.ts

```diff
@@ -2,9 +2,13 @@
  * Copyright 2020 Cognite AS
  */
 
-import { Sector, SectorQuads } from '../../models/cad/types';
+import { OperatorFunction } from 'rxjs';
+import { WantedSector } from '../../data/model/WantedSector';
+import { ParsedSector } from '../../data/model/ParsedSector';
+
+// TODO move
+export type SectorId = number;
 
 export interface Repository {
-  getDetailed: (sectorId: number) => Promise<Sector>;
-  getSimple: (sectorId: number) => Promise<SectorQuads>;
+  loadSector(): OperatorFunction<WantedSector, ParsedSector>;
```

Comment:
Would really want to see if the wanted / parsed duo could be changed, to what I am open for discussion :)

Reply:
I am also open to renaming them. Let us find some good names first, though. I feel like the `Sector` name can be used almost everywhere (for both a sector that is wanted, fetched, parsed or consumed), and so I am a bit vary of assigning one of them the final name. As mentioned, I also prefer if we can have them as different types to make sure the type system works for us.